src/main/java/state/StateFactory.java

```diff
@@ -0,0 +1,34 @@
+package state;
+
+import java.io.File;
+
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeCreatedEvent;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeSubmittedEvent;
+
+public class StateFactory {
+    public static StateAccessor getStateAccessor(File pluginDir, EiffelEvent eiffelEvent) {
+        try {
+            switch (eiffelEvent.msgParams.meta.type) {
+            case "EiffelSourceChangeCreatedEvent":
+                EiffelSourceChangeCreatedEvent eiffelSourceChangeCreatedEvent = (EiffelSourceChangeCreatedEvent) eiffelEvent;
+                String project = eiffelSourceChangeCreatedEvent.eventParams.data.gitIdentifier.repoName;
+                String branch = eiffelSourceChangeCreatedEvent.eventParams.data.gitIdentifier.branch;
+                String changeId = eiffelSourceChangeCreatedEvent.eventParams.data.change.id;
+
+                return new SourceChangeCreatedStateAccessor(pluginDir, project, branch, changeId);
+            case "EiffelSourceChangeSubmittedEvent":
+                EiffelSourceChangeSubmittedEvent eiffelSourceChangeSubmittedEvent = (EiffelSourceChangeSubmittedEvent) eiffelEvent;
+                String project1 = eiffelSourceChangeSubmittedEvent.eventParams.data.gitIdentifier.repoName;
```

Comment:
Please dont have variables named xxx1 xxx2 and so on.

Reply:
Its removed in the new commit because of some restructuring.